### PR TITLE
phantomsocks: init at unstable-2023-04-05

### PR DESCRIPTION
**Pull request by oluceps**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by ocfox** [comment]
Result of `nixpkgs-review pr 224793` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>phantomsocks</li>
  </ul>
</details>

**Review by ocfox — APPROVED** [review]

File: pkgs/tools/networking/phantomsocks/default.nix [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    rev = "7cc559b152f9d3456e630eaf9f2f689970ae238d";
+    hash = "sha256-+d/cXMZAQT9QJVSlLr4Tp19HpbRMl8jYUO8EPZFVarU=";
+  };
+
+  vendorHash = "sha256-c0NQfZuMMWz1ASwFBcpMNjxZwXLo++gMYBiNgvT8ZLQ=";
+
+  meta = with lib;{
+    homepage = "https://github.com/macronut/phantomsocks";
+    description = "A cross-platform proxy client/server for Linux/Windows/macOS";
+    license = licenses.lgpl3;
```
Comment (NickCao) [inline pkgs/tools/networking/phantomsocks/default.nix:22]: One more nitpick: `lgpl3` is deprecated, please choose `lgpl3Plus` or `lgpl3Only` after carefully inspecting the upstream license.

**Review by NickCao — COMMENTED** [review]
Not exactly happy with the `description` as it contains little information, but looks like upstream do intend to do so.

**Comment by oluceps** [comment]
> 

[This website](https://zh.wikibooks.org/wiki/Phantomsocks) provides more details. Has extended to `longDescription` now.

**Comment by NickCao** [comment]
Result of `nixpkgs-review pr 224793` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>phantomsocks</li>
  </ul>
</details>

**Pull request merged**